internal/bminventory/inventory.go

```diff
@@ -2451,9 +2456,14 @@ func (b *bareMetalInventory) updateNetworks(db *gorm.DB, params installer.Update
 				machineNetworkCidr = string(cluster.MachineNetworks[index].Cidr)
 			}
 
+			serviceNetworkCidr := ""
+			if len(cluster.ServiceNetworks) > index {
```

Comment:
That's a piggyback removing the artificial limitation of number of service network equals to the number of cluster networks. Without this, if there are more cluster networks than service networks, the code will panic.